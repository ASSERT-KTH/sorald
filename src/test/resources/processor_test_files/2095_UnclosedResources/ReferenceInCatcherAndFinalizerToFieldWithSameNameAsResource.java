/*
When the processor inlines a local variable declaration into the resources block, it also cleans
up referenses to said variable in the catchers and finalizer. This test case includes a field
with the same name as the inlined variable, and is meant to verify that we don't accidentally
delete references to this unrelated field.
 */

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class ReferenceInCatcherAndFinalizerToFieldWithSameNameAsResource {
    private int bw = 2;

    public void saveTo(File file) throws IOException {
        BufferedWriter bw = null;
        try {
            bw = new BufferedWriter(new FileWriter(file)); // Noncompliant
            bw.write("Write some stuff to file");
        } catch (IOException e) {
            e.printStackTrace();
            System.out.println(this.bw); // reference to field; should not be cleaned!
            if (bw != null) { // reference to local variable; should be cleaned!
                bw.close();
            }
        } finally {
            System.out.println(this.bw); // reference to field; should not be cleaned!
            System.out.println("done");
        }
    }
}