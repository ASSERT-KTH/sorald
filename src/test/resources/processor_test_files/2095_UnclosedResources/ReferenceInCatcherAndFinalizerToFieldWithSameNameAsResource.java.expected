/*
When the processor inlines a local variable declaration into the resources block, it also cleans
up referenses to said variable in the catchers and finalizer. This test case includes a field
with the same name as the inlined variable, and is meant to verify that we don't accidentally
delete references to this unrelated field.
 */

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class ReferenceInCatcherAndFinalizerToFieldWithSameNameAsResource {
    private int bw = 2;

    public void saveTo(File file) throws IOException {
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(file))) {
            bw.write("Write some stuff to file");
        } catch (IOException e) {
            e.printStackTrace();
            System.out.println(this.bw); // reference to field; should not be cleaned!
        } finally {
            System.out.println(this.bw); // reference to field; should not be cleaned!
            System.out.println("done");
        }
    }
}